File: solutions/strings/58.length-of-last-word.py
#
# @lc app=leetcode id=58 lang=python
#
# [58] Length of Last Word
#

# @lc code=start
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        idx = len(s)-1
        res = 0

        while idx >= 0:
            if s[idx] != " ":
                res += 1
            if res > 0 and s[idx] == " ":
                return res
            idx -= 1
        return res
        
# @lc code=end

File: solutions_java/arrays/58.length-of-last-word.java
package solutions_java.arrays;
/*
 * @lc app=leetcode id=58 lang=java
 *
 * [58] Length of Last Word
 */

// @lc code=start
class Solution {
    public int lengthOfLastWord(String s) {
        int res = 0;
        s = s.trim();

        for (int i = s.length()-1; i>=0; i--) {
          if (s.charAt(i) == ' ') {
            return res;
          }
          res += 1;
        }
        return res;
    }
}
// @lc code=end

